packages/compiler/src/init/file-templating.ts

```diff
@@ -20,19 +22,37 @@ export type FileTemplatingContext = Omit<InitTemplate, "libraries"> &
     normalizePackageName: () => (text: string, render: any) => string;
   };
 
+export interface NamingUtils {
+  readonly camelCase: () => (text: string, render: (...args: any) => string) => string;
+  readonly pascalCase: () => (text: string, render: (...args: any) => string) => string;
+  readonly kebabCase: () => (text: string, render: (...args: any) => string) => string;
+}
+
 export function createFileTemplatingContext(config: ScaffoldingConfig): FileTemplatingContext {
   return {
     ...config.template,
     ...config,
     normalizeVersion,
     toLowerCase,
     normalizePackageName,
+    naming,
   };
 }
 
 export function render(content: string, context: FileTemplatingContext): string {
   return Mustache.render(content, context);
 }
+
+const naming: NamingUtils = {
```

Comment:
Do you intend to have more general naming utils here? If not, I might call this `casing` rather than `naming`, so a template would have `casing.pascal` instead of `naming.pascalCase`.

Reply:
yeah I am not sure, as everything gets added at the root I don't know if we should have too much top level naming that could prevent user from using those names. 
Could also maybe just scope everything under `utils`

```
utils.casing.pascalCase`
```

Reply:
Ok renamed to `casing` this is actually different from the tspconfig.yaml where parmaeters are resolved at the root too. For init templates everything is kept in its structure so there isn't any unscoped user value.